### PR TITLE
OAuth: return github teams as a part of user info (enable team sync)

### PR DESCRIPTION
**Pull request by alexanderzobnin**
This PR is basic POC for getting teams from github and passing it to `ExternalUserInfo.Groups`. This achieves team sync feature for OAuth.
https://github.com/grafana/grafana-enterprise/issues/59
Based on https://github.com/grafana/grafana-private-mirror/pull/4

Github team should be specified by URL (for example, `https://github.com/orgs/grafana/teams/grafana-labs-core-devs`), but maybe it's better to use `org@slug` form, like `@grafana/grafana-labs-core-devs`.

### CODE REVIEW

File: pkg/login/social/github_oauth.go [hunk 1]
```diff
@@ -47,8 +53,13 @@ func (s *SocialGithub) IsTeamMember(client *http.Client) bool {
...
 	}
 
+	teamMembershipIds := make([]int, len(teamMemberships))
+	for i, team := range teamMemberships {
+		teamMembershipIds[i] = team.Id
+	}
+
 	for _, teamId := range s.teamIds {
-		for _, membershipId := range teamMemberships {
+		for _, membershipId := range teamMembershipIds {
```
Comment (DanCech) [inline pkg/login/social/github_oauth.go]: Any reason for creating the second array vs just:
```suggestion
		for _, membership := range teamMemberships {
		  if teamId == membership.Id {
```

File: pkg/login/social/github_oauth.go [hunk 2]
```diff
@@ -124,27 +131,27 @@ func (s *SocialGithub) FetchTeamMemberships(client *http.Client) ([]int, error)
...
-		copy(tempIds, ids)
-		ids = tempIds
+		existingRecords := len(teams)
+		tempTeams := make([]GithubTeam, (newRecords + existingRecords))
+		copy(tempTeams, teams)
+		teams = tempTeams
 
 		for i, record := range records {
-			ids[i] = record.Id
+			teams[i] = record
```
Comment (DanCech) [inline pkg/login/social/github_oauth.go]: This seems like the current version is as `i` will start from `0` every time.  We can avoid all the copying etc if we just say:
```
var teams []GithubTeam
...
for _, record := range records {
  teams = append(teams, record)
}
```

**Review by DanCech — CHANGES_REQUESTED** [review]
Looking good overall

File: pkg/login/social/github_oauth.go (hunk 1 above)
Comment (alexanderzobnin) [reply pkg/login/social/github_oauth.go]: Yea, good point! I made rough and dirty POC, so refactor is required.

File: pkg/login/social/github_oauth.go (hunk 2 above)
Comment (xlson) [reply pkg/login/social/github_oauth.go]: Using append here simplifies the code a lot.

**Review by xlson — APPROVED** [review]
LGTM!

**Comment by alexanderzobnin** [comment]
Cherry-picked Dan's auth_proxy changes as well.

**Review by DanCech — APPROVED** [review]

**Pull request merged**